### ISSUE TITLE
`ignore-missing-imports` runs afoul of `no-any-return`

### ISSUE THREAD

**Issue by allisonkarlitskaya** [python/mypy#17380]
`mypy 1.10.0 (compiled: no)` via `python3-mypy-1.10.0-1.fc40.noarch` on my Fedora 40 system running Python 3.12.3.

Consider something like this:

```python
# xyz.py
def f() -> str:
    return 'x'
```

```python
# base.py
import xyz

def abc() -> str:
    return xyz.f()
```

which you run through `mypy --strict --ignore-missing-imports`.

So far so good, everything checks out.  But let's delete `xyz.py` and do it again:

`base.py:4: error: Returning Any from function declared to return "str"  [no-any-return]`

That's because `xyz` (now missing) is assigned type `Any`, then `xyz.f` is also `Any` and the return value of `xyz.f()` is also `Any`, so you're returning `Any` from `abc()` now, earning you a `no-any-return` warning.

That's a bit unfortunate: the entire point of `--ignore-missing-imports` is not getting errors because an import is missing (and indeed, if it's not missing, there's no error).

This case seems to be relatively unique: just about anywhere else `Any` doesn't trigger errors, but here it does, and it does it in a way that conflicts with `--ignore-missing-imports`.  It always makes me wonder why this one particular case is considered particularly harmful.

So:
  - disable `no-any-return` warnings?
  - create some kind of a new "super-duper-any" which is the type of a missing import and doesn't trigger `no-any-return`?
  - something else?

Thanks!

**Comment by JelleZijlstra** [comment]
I think this is behaving as expected. `no-any-return` is a strict check that is supposed to tell you all places where you're returning something that is Any. If you don't want that, turn off the check.

**Comment by allisonkarlitskaya** [comment]
I have turned it off, but it seems a bit sub-optimal.  This issue effectively breaks the `--ignore-missing-import` feature: having a missing import triggers this error, so missing imports aren't really ignored...

At the same time, turning this flag off means that I can't benefit from it in other places...
